sdk/identity/Azure.Identity/src/AggregateAuthenticationException.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Text;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// An exception which is raised when multiple <see cref="TokenCredential"/> fail to authenticate, such as when invoked by the <see cref="ChainedTokenCredential"/> or the <see cref="DefaultAzureCredential"/>
+    /// </summary>
+    internal class AggregateAuthenticationException : AuthenticationFailedException
+    {
+        private readonly ReadOnlyMemory<object> _credentials;
+        private readonly ReadOnlyMemory<Exception> _innerExceptions;
```

Comment:
How do you iterate over these as a customer?

Reply:
The inner exception is an `AggregateException` in the case. They could cast `InnerException` to `AggregateException` and iterate through the public `InnerExceptions` property.

Reply:
Why do you need this exception type at all then?

Reply:
removed